src/poliastro/threebody/restricted.py

```diff
@@ -0,0 +1,173 @@
+# coding: utf-8
+
+import numpy as np
+from scipy.optimize import root
+
+
+def lagrange_points(r12, m1, m2):
+    """Computes the Lagrangian points using the formulation
+    in Problem 8-5 of Battin, Richard H. 'An introduction to 
+    the mathematics and methods of astrodynamics'. AIAA, 1999.
+
+    Parameters
+    ----------
+    r12 : float
+        Collinear distance
+    m1 : float
+        Mass of the main body
+    m2 : float
+        Mass of the secondary body
+
+    Returns
+    -------
+    array(float)
+        Distance of the Lagrangian points to the center,
+        projected on the collinear line
+    """
+
+    rho = r12
+
+    def eq_L1(rho2):
+        aux = 3 * rho**2 * rho2 - 3 * rho * rho2**2 + rho2**3
+        aux *= rho2**2
+        aux /= rho**3 - rho2**3
+        aux /= (rho - rho2)**2
+        aux -= m2 / m1
+        return aux
+
+    def eq_L2(rho2):
+        aux = 3 * rho**2 * rho2 + 3 * rho * rho2**2 + rho2**3
+        aux *= rho2**2
+        aux /= rho**3 - rho2**3
+        aux /= (rho + rho2)**2
+        aux -= m2 / m1
+        return aux
+
+    def eq_L3(rho1):
+        aux = 3 * rho**2 * rho1 + 3 * rho * rho1**2 + rho1**3
+        aux *= rho1**2
+        aux /= rho**3 - rho1**3
+        aux /= (rho + rho1)**2
+        aux -= m1 / m2
+        return aux
+
+    l = np.zeros((5,))
+
+    # L1
+    rho2 = root(eq_L1, 0).x
```

Comment:
I read in the documentation that the default root finding method for `scipy.optimize.root` is "MINPACK’s hybrd and hybrj routines (modified Powell method)"

https://docs.scipy.org/doc/scipy/reference/optimize.root-hybr.html.

This method works with a wide variety of functions, in particular for non differentiable functions.

https://en.wikipedia.org/wiki/Powell%27s_method

I wonder if we could get better results using `scipy.optimize.newton`, even without providing the derivative.

In fact, according to Curtis "Orbital Mechanics for Engineering Students", we should be able to use an interval method, such as `scipy.optimize.brentq`, without the need of derivatives or initial guesses:

![screenshot from 2018-08-17 19-31-38](https://user-images.githubusercontent.com/316517/44280228-6594c000-a254-11e8-98af-1704aeb9a31c.png)

Reply:
I tried `scipy.optimize.newton` but it didn't work very well, probably because of the initial guess. I'll look into `brentq`

Reply:
Notice that `brentq` will need an interval and not an initial guess, so you will need a slightly different approach

Reply:
So I looked into Curtis formulation to try to reproduce that plot, because he's formulation is adimensional and would simplify the boundary values for `brentq` or the initial estimations for `newton` and `root`. There's no problem in changing to this formulation in the deeper layer, so I might as well play with it.

I published a gist to do this and test the different root finders in `scipy.optimize`, because I'd like to use always the same.

https://gist.github.com/danbul/3e0afc694b8f4d3866a3632b4347dc03

After this quick study, I found that  `brentq` works perfectly for L2 and L3, but performs horribly for L1. For L1, only `root` with a moving initial estimation works well.

So... what do you think @Juanlu001 ?

PS: I couldn't get `newton` to work, so maybe I'm missing the hero of this story.

Reply:
Thanks for investigating this! I would also prefer that we stick to `brentq`, let me see if I can help.